workers/generic-worker/multiuser_test.go

```diff
@@ -56,3 +57,26 @@ func TestWhoAmI(t *testing.T) {
 
 	_ = submitAndAssert(t, td, payload, "completed", "completed")
 }
+
+func TestNoGenericWorkerBinaryFailsWorker(t *testing.T) {
+	setup(t)
+
+	if config.RunTasksAsCurrentUser {
+		t.Skip("Skipping since we're testing if the generic-worker binary is executable by the task user.")
+	}
+
+	exe := runtime.GenericWorkerBinary()
+	backupExe := exe + ".bak"
+	err := os.Rename(exe, backupExe)
+	if err != nil {
+		t.Fatalf("Could not move generic-worker binary to backup location: %v", err)
+	}
+	defer func() {
+		err := os.Rename(backupExe, exe)
+		if err != nil {
+			t.Fatalf("Could not move generic-worker binary to original location: %v", err)
+		}
+	}()
+
+	execute(t, INTERNAL_ERROR)
+}
```

Comment:
I don't think the generic-worker binary not existing is a likely scenario, since it would need to be removed between generic-worker starting and the generic-worker initialisation completing. Can we have a Windows helper that runs icacls to remove the Everyone execute permission, and a posix helper that runs os.Chmod to remove the "all" execute permission? This would hopefully ensure that the code that checks it is executable by the task user really is using the task user to check that, rather than if some code slipped in so that it was checking if root could execute it by mistake. This test I think wouldn't notice if generic-worker started running the check as root/LocalSystem instead of the task user.

Reply:
Yeah, good call thanks Pete.

Reply:
fwiw I think the existing `SecureFiles` function has some prior art for running icacls on Windows and os.Chmod on posix, iirc

Reply:
Yes, I plan on using that. But there's no reversal helper. I think I may just make a backup copy for the test, secure the actual binary, then test, then after it's complete, remove the secured file and restore the backup.